fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/OnlineIndexer.java

```diff
@@ -1813,25 +1862,41 @@ public OnlineIndexer build() {
             if (indexingPolicy == null) {
                 indexingPolicy = IndexingPolicy.DEFAULT;
             }
-            // Here: indexingPolicy is defined
-            Config conf = new Config(limit, maxRetries, recordsPerSecond, progressLogIntervalMillis, increaseLimitAfter, maxWriteLimitBytes);
-            return new OnlineIndexer(runner, recordStoreBuilder, index, recordTypes,
-                    configLoader, conf, syntheticIndex,
-                    useSynchronizedSession, leaseLengthMillis, trackProgress, indexingPolicy);
-        }
-
-        protected void validate() {
-            validateIndex();
-            validateLimits();
         }
 
         // Check pointer equality to make sure other objects really came from given metaData.
         // Also resolve record types to use if not specified.
         private void validateIndex() {
+            final RecordMetaData metaData = getRecordMetaData();
+            if (targetIndexes != null) {
+                // Desired state: One master index as 'index', the other targets are at 'targetIndex'.
+                targetIndexes.sort(Comparator.comparing(Index::getName));
+                if (index == null) {
+                    index = targetIndexes.get(0);
+                }
+                targetIndexes.removeIf(targetIndex -> index.getName().equals(targetIndex.getName()));
```

Comment:
I think there's an alternative here that might be a little cleaner, which is that the `.setIndex` method could be made to (1) clear out the target index list and (2) replace it with a list with a single element. Then I think the `index` field could be removed entirely, and the main index could just always be the first index in the target list.

There are maybe a few places where one would need to check to make sure that the `targetIndexes` list is empty/non-empty or has size greater than 1 (to indicate a "real" multi-target index build), but I also think there are a few special cases that could be collapsed.